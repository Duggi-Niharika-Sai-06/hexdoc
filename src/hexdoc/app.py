from flask import Flask, render_template

app = Flask(__name__)

@app.route('/')
def index():
    return render_template('table_of_contents.html.jinja')

if __name__ == '__main__':
    app.run(debug=True)
