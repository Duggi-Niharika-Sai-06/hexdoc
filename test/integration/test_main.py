# pyright: reportUnknownMemberType=false

import subprocess
from pathlib import Path
from typing import TYPE_CHECKING

import pytest
from hexdoc.cli.app import render
from pytest import MonkeyPatch, TempPathFactory
from syrupy.assertion import SnapshotAssertion

from ..conftest import list_directory

CHECK_RENDERED_FILENAMES = [
    "v/latest/main/en_us/index.html",
    "v/latest/main/en_us/index.css",
    "v/latest/main/en_us/textures.css",
    "v/latest/main/en_us/index.js",
    "v/latest/main/en_us/hexcasting.js",
    "v/latest/main/en_us/.sitemap-marker.json",
]


@pytest.fixture(scope="session", autouse=True)
def patch_versions(monkeysession: MonkeyPatch):
    # because pyright complains when we do this in the CI
    if TYPE_CHECKING:
        _hooks = None
    else:
        from hexdoc_hexcasting import _hooks

    monkeysession.setattr(_hooks, "GRADLE_VERSION", "MOD_VERSION")
    monkeysession.setattr(_hooks, "PY_VERSION", "PLUGIN_VERSION")
    monkeysession.setattr(_hooks, "FULL_VERSION", "FULL_VERSION")


@pytest.fixture(scope="session")
def app_output_dir(tmp_path_factory: TempPathFactory) -> Path:
    return tmp_path_factory.mktemp("app")


@pytest.fixture(scope="session")
def subprocess_output_dir(tmp_path_factory: TempPathFactory) -> Path:
    return tmp_path_factory.mktemp("subprocess")


@pytest.mark.hexcasting
@pytest.mark.dependency
def test_render_app_release(
    tmp_path_factory: TempPathFactory,
    snapshot: SnapshotAssertion,
    hexcasting_props_file: Path,
):
    app_output_dir = tmp_path_factory.mktemp("app")

    render(
        output_dir=app_output_dir,
        props_file=hexcasting_props_file,
        lang="en_us",
        release=True,
        branch="main",
    )

    assert list_directory(app_output_dir) == snapshot


@pytest.mark.hexcasting
@pytest.mark.dependency
def test_render_app(app_output_dir: Path, hexcasting_props_file: Path):
    render(
        output_dir=app_output_dir,
        props_file=hexcasting_props_file,
        lang="en_us",
        branch="main",
    )


@pytest.mark.hexcasting
@pytest.mark.dependency
def test_render_subprocess(subprocess_output_dir: Path, hexcasting_props_file: Path):
    cmd = [
        "hexdoc",
        "render",
        subprocess_output_dir.as_posix(),
        f"--props={hexcasting_props_file.as_posix()}",
        "--lang=en_us",
        "--branch=main",
    ]
    subprocess.run(cmd, check=True)


@pytest.mark.dependency(depends=["test_render_app", "test_render_subprocess"])
def test_file_structure(
    app_output_dir: Path,
    subprocess_output_dir: Path,
    snapshot: SnapshotAssertion,
):
    app_list = list_directory(app_output_dir)
    subprocess_list = list_directory(subprocess_output_dir)

    assert app_list == subprocess_list
    assert app_list == snapshot


@pytest.mark.dependency(depends=["test_render_app", "test_render_subprocess"])
@pytest.mark.parametrize("filename", CHECK_RENDERED_FILENAMES)
def test_files(
    filename: str,
    app_output_dir: Path,
    subprocess_output_dir: Path,
    path_snapshot: SnapshotAssertion,
):
    app_file = app_output_dir / filename
    subprocess_file = subprocess_output_dir / filename

    assert app_file == path_snapshot

    # difficult to monkeypatch versions for subprocess, so this file will be different
    if not filename.endswith(".sitemap-marker.json"):
        assert app_file.read_bytes() == subprocess_file.read_bytes()
