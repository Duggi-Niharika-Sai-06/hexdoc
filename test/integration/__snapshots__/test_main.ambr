# serializer version: 1
# name: test_file_structure
  list([
    'v',
    'v/latest',
    'v/latest/main',
    'v/latest/main/en_us',
    'v/latest/main/en_us/.sitemap-marker.json',
    'v/latest/main/en_us/apple-touch-icon-114x114.png',
    'v/latest/main/en_us/apple-touch-icon-120x120.png',
    'v/latest/main/en_us/apple-touch-icon-144x144.png',
    'v/latest/main/en_us/apple-touch-icon-152x152.png',
    'v/latest/main/en_us/apple-touch-icon-167x167.png',
    'v/latest/main/en_us/apple-touch-icon-180x180.png',
    'v/latest/main/en_us/apple-touch-icon-57x57.png',
    'v/latest/main/en_us/apple-touch-icon-60x60.png',
    'v/latest/main/en_us/apple-touch-icon-72x72.png',
    'v/latest/main/en_us/apple-touch-icon-76x76.png',
    'v/latest/main/en_us/favicon-16x16.png',
    'v/latest/main/en_us/favicon-180x180.png',
    'v/latest/main/en_us/favicon-196x196.png',
    'v/latest/main/en_us/favicon-32x32.png',
    'v/latest/main/en_us/favicon-64x64.png',
    'v/latest/main/en_us/favicon-96x96.png',
    'v/latest/main/en_us/favicon.ico',
    'v/latest/main/en_us/hexcasting.js',
    'v/latest/main/en_us/index.css',
    'v/latest/main/en_us/index.html',
    'v/latest/main/en_us/index.js',
    'v/latest/main/en_us/logo.png',
    'v/latest/main/en_us/mstile-270x270.png',
    'v/latest/main/en_us/mstile-310x150.png',
    'v/latest/main/en_us/mstile-310x310.png',
    'v/latest/main/en_us/mstile-70x70.png',
    'v/latest/main/en_us/textures.css',
  ])
# ---
# name: test_render_app_release
  list([
    'v',
    'v/MOD_VERSION',
    'v/MOD_VERSION/PLUGIN_VERSION',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/.sitemap-marker.json',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-114x114.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-120x120.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-144x144.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-152x152.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-167x167.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-180x180.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-57x57.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-60x60.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-72x72.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/apple-touch-icon-76x76.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-16x16.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-180x180.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-196x196.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-32x32.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-64x64.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon-96x96.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/favicon.ico',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/hexcasting.js',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/index.css',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/index.html',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/index.js',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/logo.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/mstile-270x270.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/mstile-310x150.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/mstile-310x310.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/mstile-70x70.png',
    'v/MOD_VERSION/PLUGIN_VERSION/en_us/textures.css',
  ])
# ---
